### ISSUE TITLE
What is the effect of batch size?

### ISSUE THREAD

**Issue by syang1993** [Rayhane-mamah/Tacotron-2#39]
Thanks for your great work.

I used your repo to train tacotron-2 with LJ-speech database with default settings. I found it learned good alignments after 6000 steps with batch size 32. But when batch size is 24 (because of the limitation of GPU memory), it cannot get a good alignment even after 23000 steps.

So I wonder how the batch size affect the performance?

**Comment by a3626a** [comment]
As batch size becomes bigger, we can expect calculated gradient to become stable. Then, we can use bigger learning rate to increase learning speed.

In general, learning rate should be halved while batch size halved.
Therefore, I suggest you to multiply 3/4 to your learning rate.

**Comment by begeekmyfriend** [comment]
![step-7000-align](https://user-images.githubusercontent.com/6031938/39956043-8e906d3c-560b-11e8-838e-f515c37eaa05.png)
It got converge at 7K steps on my test with `batch_size=16` and `r=2` as well as a bug fixed https://github.com/Rayhane-mamah/Tacotron-2/issues/35#issuecomment-388534293. However it still confuses me why the convergence comes so late (not at 2K steps). Maybe we need to adjust the learning rate?

**Comment by a3626a** [comment]
Based on my experience, reduction value affect convergence speed. Additionally, if you use smaller batch size, then you need longer training steps.

**Comment by syang1993** [comment]
@a3626a I think you're right about the relationship between gradient and batch size in theory and in my earlier experiments about Tacotron 1. But when I tried to reduce the learning rate for this repo, it also cannot get alignments after 30k steps with `r=1` and `batch_size=24`.  So I'm confused about it.

And about the reduction factor, the author of Tacotron 1 paper said it affected the convergence speed.   Maybe I should change the value of reduction value to see if it works with smaller batch size.

**Comment by syang1993** [comment]
@begeekmyfriend Thanks for you sharing. I will try your suggestion to see if it solves this. I also found the reduction value `r` affects the convergence, but I don't know how it affects.
